src/services/get-songs.ts

```diff
@@ -252,13 +252,13 @@ export default class {
       }
     }
 
-    // Get 50 random songs if many
+    // Get random songs if the playlist is larger than limit
     const originalNSongs = tracks.length;
 
-    if (tracks.length > 50) {
+    if (tracks.length > limit) {
       const shuffled = shuffle(tracks);
 
-      tracks = shuffled.slice(0, 50);
+      tracks = shuffled.slice(0, limit);
```

Comment:
I think there may have been some confusion around how the playlist limit was implemented. Currently / previously (code on main branch), we fetch all results from both YouTube and Spotify playlists. The actual limit only controls a view of the data (implemented here) rather than having any effect on the actual fetching of playlist items themselves.